include/alp/reference/vector.hpp

```diff
@@ -145,7 +147,8 @@ namespace alp {
 				 * \warning Avoid the use of this constructor within performance critical
 				 *          code sections.
 				 */
-				Vector( const size_t length, const size_t cap = 0 ) : n( length ), cap( std::max( length, cap ) ), initialized( false ) {
+				Vector( const size_t length, const size_t cap = 0 ) :
+					n( length ), cap( std::max( length, cap ) ), initialized( false ), should_free_data( true ) {
```

Comment:
With `should_free_data`, are we creating a shared_ptr-like logic here? Perhaps we should just switch to smart pointers?